src/interface/src/app/treatments/treatments.state.ts

```diff
@@ -61,65 +63,83 @@ export class TreatmentsState {
     return this._scenarioId;
   }
 
-  setInitialState(opts: {
-    scenarioId: number;
-    treatmentId: number;
-    projectAreaId: number | undefined;
-  }) {
-    this._scenarioId = opts.scenarioId;
-    this._treatmentPlanId = opts.treatmentId;
-    this._projectAreaId$.next(opts.projectAreaId);
+  /**
+   * Uses route data to load summary and treatment plan
+   * returns true when both are loaded.
+   * @param data
+   */
+  loadTreatmentByRouteData(data: TreatmentRoutingData) {
```

Comment:
might be easier to diff this side-by-side, git is scrambling all the methods by line number